app-modules/timeline/src/Filament/Pages/Timeline.php

```diff
@@ -89,15 +90,21 @@ public function getSubNavigation(): array
 
     public static function shouldRegisterNavigation(array $parameters = []): bool
     {
+        if (auth()->user()->cannot('engagement.view-any')) {
+            return false;
+        }
```

Comment:
We might want to introduce a custom permission for this because the timeline itself consists of many different types of records, and it's possible a user can view some records but not all.